src/main/java/net/shadowmage/ancientwarfare/npc/item/ItemNpcSpawner.java

```diff
@@ -68,6 +72,19 @@ public String getUnlocalizedName(ItemStack stack) {
 		return super.getUnlocalizedName(stack);
 	}
 
+	@Override
+	public String getItemStackDisplayName(ItemStack stack) {
+		return getNpcDisplayNameFromTag(stack).isPresent() ? getNpcDisplayNameFromTag(stack).get() : super.getItemStackDisplayName(stack);
+	}
+
+	private static Optional<String> getNpcDisplayNameFromTag(ItemStack stack) {
+		if (stack.hasTagCompound() && stack.getTagCompound().hasKey(NPC_STORED_DATA_TAG) &&
+				stack.getTagCompound().getCompoundTag(NPC_STORED_DATA_TAG).hasKey("name")) {
+			return Optional.of(stack.getTagCompound().getCompoundTag(NPC_STORED_DATA_TAG).getString("name"));

```

Comment:
it would make sense to encapsulate all usages of `stack.getTagCompound()` into a variable (let's say "tag") to shorten the code and make it more readable.